src/main/java/org/apache/commons/csv/ExtendedBufferedReader.java

```diff
@@ -35,7 +34,7 @@
  * {@link #read()}. This reader also tracks how many characters have been read with {@link #getPosition()}.
  * </p>
  */
-final class ExtendedBufferedReader extends BufferedReader {
+final class ExtendedBufferedReader extends BufferedReaderFork {
```

Comment:
@mmstan,

If your goal is to use a `Reader` that does not use synchronized blocks, why don't you replace `BufferedReader` with `Reader`? As far as I can tell, `ExtendedBufferedReader` implements almost all the methods used by `Lexer`.

`BufferedReaderFork` is rather a no-go: it would require a lot of maintenance work over the years, whereas classes such as `Reader` and `BufferedReader` are maintained by OpenJDK.